code/datums/cargo.dm

```diff
@@ -69,7 +69,13 @@
 	data["price_modifier"] = price_modifier
 	return data
 
+/datum/cargo_supplier/proc/get_total_price_coefficient()
+	var/final_coef = price_modifier
+	if(SSatlas.current_sector)
+		if(short_name in SSatlas.current_sector.cargo_price_coef)
+			final_coef = SSatlas.current_sector.cargo_price_coef["Name"] * price_modifier
```

Comment:
You are checking for the existence of the key with the value `short_name` in `cargo_price_coef`.

To calculate the `final_coef` you are are getting the key with the name "Name" from the array `cargo_price_coef`

The key used to access the cargo_price_coeff should most likely be `short_name`